Source/WebKit/UIProcess/WebPageProxy.messages.in

```diff
@@ -302,6 +302,8 @@ messages -> WebPageProxy {
 
     # Drag and drop messages
 #if PLATFORM(COCOA) && ENABLE(DRAG_SUPPORT)
+    PropagateDragAndDrop(WebCore::FrameIdentifier frameID, struct WebCore::RemoteUserInputEventData remoteUserInputData) -> (bool handled)
+    FetchSandboxExtensionsForDragAction(WebCore::FrameIdentifier frameID) -> (WebKit::SandboxExtensionHandle sandboxExtensionHandle, Vector<WebKit::SandboxExtensionHandle> sandboxExtensionsForUpload)
```

Comment:
It seems unfortunate that we now need this additional IPC message at the end of the drag and drop, even when site isolation is disabled. Any frame can claim they need the sandbox extensions during hit testing so it's not obvious to me this is better than sending the sandbox extensions to each WP.

Reply:
I've done it both ways https://github.com/WebKit/WebKit/pull/28258#discussion_r1605309794